notebook/notebookapp.py

```diff
@@ -2294,16 +2325,36 @@ def list_running_servers(runtime_dir=None):
             with io.open(os.path.join(runtime_dir, file_name), encoding='utf-8') as f:
                 info = json.load(f)
 
-            # Simple check whether that process is really still running
+            # Actively check whether that process is really available via an HTTP request
             # Also remove leftover files from IPython 2.x without a pid field
-            if ('pid' in info) and check_pid(info['pid']):
-                yield info
-            else:
-                # If the process has died, try to delete its info file
+            response = kernel_request(info, path=url_path_join(info.get('base_url') or '/','/api'))
+            def flush_info_file():
                 try:
                     os.unlink(os.path.join(runtime_dir, file_name))
                 except OSError:
-                    pass  # TODO: This should warn or log or something
+                    pass
+
+            version_dict = {}
+            try:
+                version_dict.update(json.loads(response.body.decode()))
+            except:
+                flush_info_file()
+            else:
+                received_version = version_dict.get('version', '0.0.0')
+                msg_special = "Jupyter Notebook HIGHER VERSION detected: %s, current:%s" %(
```

Comment:
```suggestion
                msg_special = "WARNING: A different Jupyter Notebook version was detected: %s, current: %s" %(
```

Reply:
we encourage user's hack version ;-)
"higher" is better that "different" for suck kind of users;
higher version !